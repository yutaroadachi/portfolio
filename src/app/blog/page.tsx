import {
  getPersonalBlogList,
  getTechBlogList,
  PersonalBlogList,
  TechBlogList,
} from '@/features/blog/blog-list'
import { ProfileCard } from '@/features/profile/profile-card'
import { generateOpenGraph, generateTitle } from '@/utils/meta/generator'
import { Suspense } from 'react'

const title = 'ブログ'
const description = 'Yutaro Adachi(Frontend Engineer)のブログです。'
export const metadata = {
  title: generateTitle(title),
  description,
  openGraph: generateOpenGraph(title, description),
}

export const revalidate = 300

export default async function BlogListPage() {
  const techBlogListData = getTechBlogList()
  const personalBlogListData = getPersonalBlogList()

  const techBlogList = await techBlogListData

  return (
    <div className="space-y-8">
      <section>
        <h1 id="blog-list">ブログ</h1>
      </section>
      <TechBlogList techBlogList={techBlogList} />
      <Suspense fallback={<h2>Loading…</h2>}>
        {/* @ts-expect-error Server Component */}
        <PersonalBlogListData personalBlogListData={personalBlogListData} />
      </Suspense>
      <ProfileCard />
    </div>
  )
}

const PersonalBlogListData = async ({
  personalBlogListData,
}: {
  personalBlogListData: ReturnType<typeof getPersonalBlogList>
}) => {
  const personalBlogList = await personalBlogListData

  return <PersonalBlogList personalBlogList={personalBlogList} />
}
