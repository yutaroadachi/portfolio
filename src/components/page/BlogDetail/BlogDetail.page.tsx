import { Box } from '@chakra-ui/react'
import React, { ComponentProps } from 'react'
import { MetaTag } from 'src/components/common'
import { AppLayout } from 'src/components/layout'
import { BlogDetail } from './BlogDetail'

export type BlogDetailPageProps = ComponentProps<typeof BlogDetail>

export const BlogDetailPage = ({ blog }: BlogDetailPageProps) => {
  return (
    <AppLayout>
      <MetaTag
        title={blog.title}
        description={blog.description ?? blog.title}
      />
      <Box
        w={{ base: 'full', lg: 'container.md' }}
        px={{ base: 4, lg: 0 }}
        mx="auto"
      >
        <BlogDetail blog={blog} />
      </Box>
    </AppLayout>
  )
}
