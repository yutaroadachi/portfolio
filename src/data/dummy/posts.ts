import { RssFeedItem } from 'src/types/rss'

export const dummyPosts: RssFeedItem[] = Array(13)
  .fill(0)
  .map((_, idx) => ({
    content: '',
    contentSnippet: '',
    creator: '',
    'dc:creator': '',
    enclosure: {
      url: '',
      length: '',
      type: '',
    },
    guid: '',
    isoDate: '',
    link: 'http://example.com',
    pubDate: 'Sat, 28 Aug 2021 00:00:00 GMT',
    title: `テスト記事${idx + 1}`.repeat(idx + 1),
  }))
