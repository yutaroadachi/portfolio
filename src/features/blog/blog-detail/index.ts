export * from './get-personal-blog-detail'
export * from './personal-blog-detail'
