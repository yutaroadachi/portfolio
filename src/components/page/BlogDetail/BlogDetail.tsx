import NextLink from 'next/link'
import { BiChevronRight } from 'react-icons/bi'
import { Biography } from 'src/components/domain/Biography'
import { PATH } from 'src/constants/path'
import { formatYYYYMMDD } from 'src/lib/format/date-format'
import { PersonalBlogDetail } from 'src/types/blog'

export type BlogDetailProps = {
  blog: PersonalBlogDetail
}

export const BlogDetail = ({ blog }: BlogDetailProps) => {
  return (
    <div className="space-y-8">
      <div>
        <nav aria-label="breadcrumb" className="text-sm text-c-link">
          <ol className="flex items-center space-x-1">
            <li>
              <NextLink href={PATH.blog}>ブログ一覧</NextLink>
            </li>
            <BiChevronRight className="text-gray-500" />
            <li>
              <NextLink href={`/blog/${blog.id}`}>ブログ詳細</NextLink>
            </li>
          </ol>
        </nav>
        <h1 id="blog-detail">{blog.title}</h1>
        <div className="text-sm text-gray-500">
          {`${formatYYYYMMDD(
            new Date(blog.publishedAt ?? blog.createdAt)
          )}に投稿`}
        </div>
      </div>
      <div
        dangerouslySetInnerHTML={{
          __html: `${blog.body}`,
        }}
        className="blog-detail card p-8"
      />
      <div className="center">
        <NextLink href={PATH.blog} className="text-c-link">
          ブログ一覧へ
        </NextLink>
      </div>
      <Biography />
    </div>
  )
}
