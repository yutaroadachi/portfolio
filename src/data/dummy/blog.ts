import { BlogDetail, BlogList } from 'src/types/blog'

export const dummyBlogList: BlogList = Array(13)
  .fill(0)
  .map((_, idx) => ({
    id: (idx + 1).toString(),
    createdAt: '2021-10-31T00:00:00.0000',
    updatedAt: '2021-10-31T00:00:00.0000',
    publishedAt: '2021-10-31T00:00:00.0000',
    revisedAt: '2021-10-31T00:00:00.0000',
    title: `テスト個人ブログ${idx + 1}`,
    body: `<p>テスト個人ブログ${
      idx + 1
    }です。<br><br><a href="http://example.com" target="_blank" rel="noopener noreferrer">テストリンク</a></p>`,
    description: `テスト個人ブログ${idx + 1}です。`,
  }))

export const dummyBlogDetail: BlogDetail = {
  id: '1',
  createdAt: '2021-10-31T00:00:00.0000',
  updatedAt: '2021-10-31T00:00:00.0000',
  publishedAt: '2021-10-31T00:00:00.0000',
  revisedAt: '2021-10-31T00:00:00.0000',
  title: 'テスト個人ブログ1',
  body: '<p>テスト個人ブログ1です。<br><br><a href="http://example.com" target="_blank" rel="noopener noreferrer">テストリンク</a></p>',
  description: 'テスト個人ブログ1です。',
}
