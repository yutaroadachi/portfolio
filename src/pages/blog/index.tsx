import { GetStaticProps } from 'next'
import React from 'react'
import { BlogList, BlogListProps } from 'src/components/page/BlogList'
import { createGetLayout } from 'src/components/ui/Layout'
import { MetaTag } from 'src/components/ui/MetaTag'
import { getPersonalBlogList } from 'src/lib/microCMS/client'
import { getZennFeed } from 'src/lib/rss/client'

export const getStaticProps: GetStaticProps<BlogListProps> = async () => {
  const [techBlog, personalBlog] = await Promise.all([
    getZennFeed(),
    getPersonalBlogList(),
  ])

  return {
    props: {
      techBlog: techBlog,
      personalBlog: personalBlog,
    },
    revalidate: 5 * 60,
  }
}

export default function BlogListPage({
  techBlog,
  personalBlog,
}: BlogListProps) {
  return (
    <>
      <MetaTag
        title="ブログ"
        description="Yutaro Adachi(ソフトウェアエンジニア)のブログです。"
      />
      <BlogList techBlog={techBlog} personalBlog={personalBlog} />
    </>
  )
}

BlogListPage.getLayout = createGetLayout()
