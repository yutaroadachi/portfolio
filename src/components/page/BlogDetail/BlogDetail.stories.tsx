import { Meta, Story } from '@storybook/react'
import React from 'react'
import { dummyPersonalBlogDetail } from 'src/data/dummy/blog'
import { BlogDetail, BlogDetailProps } from './BlogDetail'

export default {
  title: 'page/BlogDetail',
  component: BlogDetail,
  args: {
    blog: dummyPersonalBlogDetail,
  },
} as Meta<BlogDetailProps>

type Template = Story<BlogDetailProps>

const Template: Template = (args) => <BlogDetail {...args} />

export const Default: Template = Template.bind({})
