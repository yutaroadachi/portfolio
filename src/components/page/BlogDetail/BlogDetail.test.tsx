import React from 'react'
import { dummyBlogDetail } from 'src/data/dummy/blog'
import { render } from 'src/__tests__/utils'
import { BlogDetail } from './BlogDetail'

test('BlogDetail', () => {
  const { container } = render(<BlogDetail blog={dummyBlogDetail} />)
  expect(container).toMatchSnapshot()
})
