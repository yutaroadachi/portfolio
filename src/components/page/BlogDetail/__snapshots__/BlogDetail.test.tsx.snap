// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogDetail 1`] = `
<div>
  <div
    class="chakra-stack css-egoftb"
  >
    <div
      class="css-0"
    >
      <h2
        class="css-553wiy"
        id="blog-detail"
      >
        テスト個人ブログ1
      </h2>
      <p
        class="css-1e82jug"
      >
        Posted at 2021/10/31
      </p>
    </div>
    <div
      class="blog-detail css-0"
    >
      <p>
        テスト個人ブログ1です。
        <br />
        <br />
        <a
          href="http://example.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          テストリンク
        </a>
      </p>
    </div>
  </div>
</div>
`;
